PSReadLine/SamplePSReadlineProfile.ps1

```diff
@@ -127,7 +127,12 @@ Set-PSReadlineKeyHandler -Key '"',"'" `
     $cursor = $null
     [Microsoft.PowerShell.PSConsoleReadLine]::GetBufferState([ref]$line, [ref]$cursor)
 
-    if ($line[$cursor] -eq $key.KeyChar) {
+    $quoteNumber = Select-String -InputObject $line -Pattern $key.KeyChar -AllMatches
+    if ($quoteNumber.Matches.Count % 2 -eq 1) {
+        # Oneven amount of quotes, put just one quote
+        [PSConsoleUtilities.PSConsoleReadline]::Insert($key.KeyChar)
```

Comment:
`PSConsoleUtilities` should be changed to `Microsoft.PowerShell`.  Also, Select-String is a bit overkill.   I think this should do:
```powershell
    $quoteCount = @([regex]::Matches($line, $key.KeyChar)).Length
    if ($quoteCount % 2 -eq 1) {
        # On odd number of quotes, insert just one quote
        [Microsoft.PowerShell.PSConsoleReadline]::Insert($key.KeyChar)
```